File: src/main/java/programmers/Lim42842.java
package programmers;

public class Lim42842 {
    public int[] solution(int brown, int yellow) {
        final int sum = brown / 2 + 2;
        for (int i = 3; i <= sum - 3; i++) {
            if ((i - 2) * (sum - i - 2) == yellow) {
                return new int[]{sum - i, i};
            }
        }

        return new int[]{1, 1};
    }
}

File: src/main/java/programmers/part2/Lim42842.kt
package programmers.part2

class Lim42842 {
    fun solution(brown: Int, yellow: Int): IntArray {
        val sum = brown / 2 + 2 // 가로와 세로의 합
        for (i in 3..sum - 3) {
            if ((i - 2) * (sum - i - 2) == yellow) {
                return intArrayOf(sum - i, i)
            }
        }
        return intArrayOf()
    }
}
